### PR TITLE
apt consistency

### PR DESCRIPTION
**Pull request by nbrady-techempower**
Let's keep `apt` consistent to really utilize docker caching

This needs the full travis run, I did a lot of regex replacing, so I want to make sure I didn't goof anything.

A couple of notes:

  - `apt update -yqq` doesn't need `> /dev/null`, it's quiet except for one warning
  - let's leave `apt update -yqq` on its own line, and let's try to keep it at the top of the dockerfile where applicable going forward
  - added `> /dev/null` to all apt installs

### CODE REVIEW

**Comment by msmith-techempower** [comment]
Let's let this Travis run play out; it will shed light on any areas where `/dev/null` is a problem.

... or push a new commit while I'm typing this, that's good, too.

**Comment by nbrady-techempower** [comment]
🚤

**Comment by michaelhixson** [comment]
I strongly prefer multiple `RUN` statements over combining with `&&` in general, but I noticed Docker specifically recommends combining apt `update` and `install` commands because of the interaction with caching.

https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#run

> Always combine RUN apt-get update with apt-get install in the same RUN statement. For example:
```
    RUN apt-get update && apt-get install -y \
        package-bar \
        package-baz \
        package-foo
```
> Using apt-get update alone in a RUN statement causes caching issues and subsequent apt-get install instructions fail.

I don't understand the potential issue well enough to say "we must do this", it just seemed worth mentioning.

**Comment by nbrady-techempower** [comment]
> Using apt-get update alone in a RUN statement causes caching issues and subsequent apt-get install instructions fail.

That's interesting. We already have these separated in a number of places. I don't think we've seen anything fail in practice because of this?

**Comment by michaelhixson** [comment]
Yeah... I don't even know what that documentation I quoted means by "fail".  The example given afterwards doesn't sound like a failure.  I'm fine with ignoring their advice about `update` and `install` for now.

**Comment by joanhey** [comment]
Use `apt` or `apt-get` is the same, and is better use one for consistency.

I'm not a fan of `/dev/null` too. But the impact in extra noise it's very big.

Before every framework had a travis job, was OK.
Now that all are joined for language, it's a problem this extra verbosity without any benefit.

### With `/dev/null`
https://travis-ci.org/TechEmpower/FrameworkBenchmarks/jobs/373893021
Fast and easy to find a problem. And have extra place for more frameworks.

### Without `/dev/null`
https://travis-ci.org/TechEmpower/FrameworkBenchmarks/jobs/374575798
If fail a framework after line 10.000, check the log:
https://api.travis-ci.org/v3/job/374575798/log.txt

The log is unreadable.
First check which framework don't pass.
Later try to find the problem in that log.

**Comment by dantti** [comment]
I still think this should be in a base.sh instead of RUN with apt calls, this is error prone once one "base" package needs to be added or new FW tests get in.

**Comment by nbrady-techempower** [comment]
@dantti I'm not sure what you mean by a base sh file? one for all frameworks? we're avoiding that now so that each tests dockerfile can be run independently of the toolset. In other words, all you would need is docker installed, and that framework directory, and you could build the test from that directory context without the toolset.

File: frameworks/C/h2o/start-servers.sh [hunk 1]
```diff
@@ -35,7 +35,7 @@ build_h2o_app()
 run_curl()
 {
 	for ((i = 0; i < 10; i++)); do
-		curl "${H2O_APP_PROFILE_URL}/$1" > /dev/null 2>&1
```
[the remark this answers is not in the document]
Comment (nbrady-techempower) [reply frameworks/C/h2o/start-servers.sh]: This was unintentional. I think it got caught up in a sweeping regex change. Nice spot!

**Comment by nbrady-techempower** [comment]
I'm going to close this and just try and keep consistency in PR reviews for now

**Pull request closed without merging**